### PR TITLE
Feature/condo/doma 1272/validate i18n translations

### PR DESCRIPTION
**Pull request by AntonAL**
Validating i18n translation keys

Displays difference between keys between i18n translation files `ru.json` и `en.json`.
Supports any count of language files (for the future).
Get's started on `pre-commit` Git hook.
Uses paired comparing and exits with error code when first pair of languages has differences. It prevents commit in Git.

It loads files from `lang` folder of current working directory (cwd)
and compares it pair-by-pair.
It exits with code 1 in case of spotted differences in keys set and order.

## Validation rules

Translation files should match following rules:
1. All supported languages should have the same set of keys
2. Each key should be on the same line in JSON-file for each language

## Use-cases

Typical use cases would be to launch this util:
- As a pre-commit hook locally, to get instant feedback, containing inconsistencies,
  made by changes, being committed
- As part of a CI/CD workflow

### Locally as a pre-commit hook

To use it as a pre-commit hook, add following entry into `script` section of repository's `package.json`:

```json
"lint-i18n-translations": "node ./../../bin/lint-i18n-translations.js",
```

In `.husky/pre-commit` add a launch command, like following, supposing, that your repo
is named `your_repo` and located in `apps` folder:

```
yarn workspace @app/your_repo lint-i18n-translations
```

Update pre-commit hooks, for example, when husky is in `prepare` script, run

```shell
npm run prepare
```

### As part of CI/CD workflow

Add following command to appropriate workflow configuration file:

```
yarn workspace @app/condo lint-i18n-translations
```

### CODE REVIEW

**Comment by pahaz** [comment]
you can add a new lint step here: https://github.com/open-condo-software/condo/blob/master/.github/workflows/nodejs.codestyle.yml#L25

**Review by Dimitreee — APPROVED** [review]

**Review by SavelevMatthew — APPROVED** [review]
LGTM

File: bin/lint-i18n-translations.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
+    Validating i18n translation keys
+    Translation files should match following rules:
+    1. All supported languages should have the same set of keys
+    2. Each key should be on the same line in JSON-file for each language
+*/
+const fs = require('fs')
+const { diffLinesUnified, diffLinesRaw } = require('jest-diff')
+
+const langDir = 'apps/condo/lang'
```
Comment (pahaz) [inline bin/lint-i18n-translations.js:10]: it's more reusable if you can detect this path or discover it by working directory. It's help to use this file for others applications

Comment (pahaz) [reply bin/lint-i18n-translations.js:10]: You can use something similar to `lint-schema.js`

File: bin/lint-i18n-translations.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    const lang = fileName.match(/(\w+)\.json/)[1]
+    const filePath = `${langDir}/${fileName}`
+    console.log(`Load file ${filePath}`)
+
+    let data
+    try {
+        data = fs.readFileSync(filePath, 'utf8')
+    } catch (e) {
+        console.log(`Error reading file "${fileName}"`)
+        console.error(e)
```
Comment (pahaz) [inline bin/lint-i18n-translations.js:24]: process.exit(1)

Comment (pahaz) [reply bin/lint-i18n-translations.js:24]: is you can't read the file it means that something gets wrong.

File: bin/lint-i18n-translations.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    }
+    if (!data) {
+        return
+    }
+    let parsedData
+    try {
+        parsedData = JSON.parse(data)
+    } catch (e) {
+        console.log(`Error reading file "${fileName}"`)
+        console.error(e)
```
Comment (pahaz) [inline bin/lint-i18n-translations.js:34]: process.exit(1)

File: bin/lint-i18n-translations.js (hunk 2 above)
Comment (AntonAL) [reply bin/lint-i18n-translations.js:24]: Added

File: bin/lint-i18n-translations.js (hunk 3 above)
Comment (AntonAL) [reply bin/lint-i18n-translations.js:37]: Added

Comment (AntonAL) [reply bin/lint-i18n-translations.js:40]: Added

**Pull request merged**